server-worker/src/main/scala/com/aliyun/emr/rss/service/deploy/worker/Controller.scala

```diff
@@ -111,6 +113,13 @@ private[deploy] class Controller(
       splitMode: PartitionSplitMode,
       partitionType: PartitionType): Unit = {
     val shuffleKey = Utils.makeShuffleKey(applicationId, shuffleId)
+    if (shutdown.get()) {
+      val msg = "Current worker is shutting down!"
+      logError(s"[handleReserveSlots] $msg")
+      context.reply(ReserveSlotsResponse(StatusCode.ReserveSlotFailed, msg))
```

Comment:
Shall we add a new status code like WorkerShuttingDown to be more specific?

Reply:
> Shall we add a new status code like WorkerShuttingDown to be more specific?

Prefer to do this in a refactor pr for all such cases as we have discussed before